### PR TITLE
add docker support

### PR DESCRIPTION
**Pull request by darwin**

### CODE REVIEW

**Comment by bitromortac** [comment]
Thank you very much for the pull request, it's very much appreciated! I'll have a closer look into it in the next days (I'm not so knowledgable in docker things).

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# we have a really simple setup here
+# we use docker-compose only as a convenient way to specify docker build parameters via a yaml file
+# you could as well use Dockerfile directly with `docker build` and config passed via command-line args
+#
+# tips:
+#   - to run lndmanage from docker, you can use our wrapper script ./lndmanage.sh
+#   - see other scripts in this folder, also check the docker section in the main readme
+
+version: '3.7'
```
Comment (bitromortac) [inline docker/docker-compose.yml:9]: Would it be problematic to lower the version? From the Ubuntu standard installation, I can run version 3.5.

**Review by bitromortac — COMMENTED** [review]
Thanks again! I tried to go through the Dockerfile and tried to build the image. As I understand, this is intended to be run on the host that also runs the lnd node, which makes sense for having everything in one place.

I was running this however remotely and therefore didn't have any ```~/.lnd folder```. So I tried to mount some other path (say ```/home/user``` to ```/root/.lnd```), just to test if it gets mounted. When I do the inspection, however there is no ```/root/.lnd``` folder, nor is there a ```/root/lndmanage/cache``` folder, which should be there anyhow, right? In the description of this project, we state that lndmanage is optimized for remote control. Do you think there would be demand for this use case in connection with a docker image?

File: docker/docker-compose.yml (hunk 1 above)
Comment (darwin) [reply docker/docker-compose.yml:9]: Sure, feel free to downgrade it. It should work the same I believe.

**Comment by darwin** [comment]
Well, I used it in both scenarios. Having all local is easy (and the `config_sample.ini` does not need any changes) . 

In remote scenario, I created a local folder which mimics the real folder structure:
```
> export LND_HOME=/Users/darwin/lnd-data
>ls -T /Users/darwin/lnd-data
/Users/darwin/lnd-data/
├── data/
│  └── chain/
│     └── bitcoin/
│        └── mainnet/
│           └── admin.macaroon
└── tls.cert
```
Alternatively user could put both files into a single directory and then tweak `tls_cert_file` and `admin_macaroon_file` in `docker/lndmanage/home/config_sample.ini`.

As for inspection. What you describe is consistent with what I would expect. And I agree that it might be a bit confusing for someone new to docker (I'm open to changes).

Mounting is defined in `docker-compose.yml` file. So only containers created via docker-compose are properly setup with all options from that file. When you run `./inspect.sh` it creates a new container directly via docker's `run` command. That is useful just for seeing what is inside the image, but not useful for inspecting full picture. To see what you expected, you would have to do `docker-compose up`, this would create a new container named `lndmanage` running in "background", and then while it is running, you could `./inspect.sh exec` launch in another terminal session. This would launch a new shell process in already running container, where you would see mounted volumes, env vars, etc.

If you look at `./lndmanage.sh` implementation, you can see that I check if `lndmanage` container is already running and do either run or exec depending on situation. Now when I think about it, I should have implemented `./inspect.sh` using `docker-compose` instead of low-level `docker` command.

**Comment by darwin** [comment]
I have just rebased the PR. Also improved the inspect.sh script.

**Comment by bitromortac** [comment]
Thank you for the work and explanations, definiteley learned something!

**Pull request merged**